### PR TITLE
first pull request - karen's portfolio

### PR DESCRIPTION
**Pull request by karenav**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -13,16 +13,45 @@
...
+  const factContainer = document.getElementById('fact-container');
+  factContainer.innerText = fact;
 }
+
+/**
+ * Creates a semi-random story and adds it to the page.
+ */
+function randomizeStory(){
+    var inputFromUser = document.getElementById('inputName').value;
+    var inputName = (inputFromUser=='') ? "Karen" : inputFromUser;
```
Comment (szwbar) [inline portfolio/src/main/webapp/script.js:35]: Add whitespace before and after any punctuation mark:

`inputFromUser == ''`

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -13,16 +13,45 @@
...
+
+    text = text + "This is a story about a " + getRandomItemFromArray(characterList) + 
+        " called " + inputName + ". " + inputName + " had to find the magical " + 
+        getRandomItemFromArray(objectsList) + " in order to cure their beloved sick " + 
+        getRandomItemFromArray(characterList) + ". " + inputName + 
+        " sacrifised everything for it, including losing their favourite " + 
+        getRandomItemFromArray(objectsList) + 
+        ". But it was all worth it, and they lived happily ever after."
+
+    document.getElementById("storyBox").innerHTML=text;
```
Comment (szwbar) [inline portfolio/src/main/webapp/script.js:48]: Same here (` = text`)

Comment (szwbar) [reply portfolio/src/main/webapp/script.js:48]: And other places. :)

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -13,16 +13,45 @@
...
+document.getElementById("storyBox").style.visibility='visible';
+}
+
+/**
+* returns a semi-ramdom item from the given array.
```
Comment (szwbar) [inline portfolio/src/main/webapp/script.js:53]: nit: capital letter (Returns)

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -13,16 +13,45 @@
...
+*/ 
+function getRandomItemFromArray(array){
```
Comment (szwbar) [inline portfolio/src/main/webapp/script.js:55]: whitespace after `array) `

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -13,16 +13,45 @@
...
+    return array[Math.floor(Math.random()*(array.length))];
```
Comment (szwbar) [inline portfolio/src/main/webapp/script.js:56]: whitespaces in the multiplication.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -2,8 +2,47 @@
...
   margin-right: auto;
   width: 650px;
+  padding: 5px;
 }
 
 #greeting-container {
   margin-top: 20px;
 }
+
+body {background-color: rgb(196, 238, 243);
```
Comment (szwbar) [inline portfolio/src/main/webapp/style.css:12]: Please structure your CSS as follows:
```css
body {
...
}
```

Also other CSS rules.

File: portfolio/src/main/webapp/style.css [hunk 7]
```diff
@@ -2,8 +2,47 @@
...
+  color: white;
+  padding: 8px 20px;
+  text-align: center;
+  display: inline-block;
+  font-size: 13px;
+  cursor: pointer;
+  font-family: Tahoma, Geneva, sans-serif;
+}
+
+input[type="text"] {
```
Comment (szwbar) [inline portfolio/src/main/webapp/style.css:30]: Shouldn't you define a class for this?

File: portfolio/src/main/webapp/style.css [hunk 8]
```diff
@@ -2,8 +2,47 @@
...
+
+#storyBox {
+    background-color: rgb(255, 173, 158); 
+    visibility: hidden;
+    padding: 10px;
+    width: 600px;
+    height:auto;
+}
+
+/* #randomStory {
```
Comment (szwbar) [inline portfolio/src/main/webapp/style.css:43]: Remove?

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (karenav) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (karenav) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (karenav) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (karenav) [reply portfolio/src/main/webapp/script.js]: done

File: portfolio/src/main/webapp/style.css (hunk 6 above)
Comment (karenav) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/style.css (hunk 7 above)
Comment (karenav) [reply portfolio/src/main/webapp/style.css:32]: I saw examples of this online, so it seems it's a common practice

File: portfolio/src/main/webapp/style.css (hunk 8 above)
Comment (karenav) [reply portfolio/src/main/webapp/style.css]: done

File: portfolio/src/main/webapp/style.css (hunk 7 above)
Comment (szwbar) [reply portfolio/src/main/webapp/style.css:32]: The problem with this approach is that if you have a big website with multiple input text boxes, all of them will be styled according to this CSS rule. I think that usually having a specific rule for each element (or element group) is a better practice. 

You can leave as-is for the starter projects, in any case.

**Pull request merged**